### PR TITLE
Add user types to process and etcd service

### PR DESCRIPTION
**Pull request by moelsayed**

### CODE REVIEW

**Comment by superseb** [comment]
LGTM

File: apis/management.cattle.io/v3/rke_types.go [hunk 1]
```diff
@@ -244,6 +244,11 @@ type ETCDService struct {
 	Key string `yaml:"key" json:"key,omitempty"`
 	// External etcd prefix
 	Path string `yaml:"path" json:"path,omitempty"`
+	// UID to run etcd container as
+	UID int `yaml:"uid" json:"uid,omitempty"`
```
Comment (alena1108) [inline apis/management.cattle.io/v3/rke_types.go]: can you provide an example of gid/uid

Comment (alena1108) [reply apis/management.cattle.io/v3/rke_types.go]: We'll need to mention it in the steps for qa as well. Also can you describe why we can't use a string type here

Comment (moelsayed) [reply apis/management.cattle.io/v3/rke_types.go]: On a typical ubuntu server:
```
$ id ubuntu 
uid=1000(ubuntu) gid=1000(ubuntu)
```
However, for this particular use case, the user will need to create a specific user for etcd on the node before provisioning:
```
# useradd etcd 
# id etcd
uid=1001(etcd) gid=1001(etcd) groups=1001(etcd)
```

The string username/group are just user friendly representations of the actual integer uid/gid used by the system, the mapping is done in the /etcd/passed file. If we use  username/group in this case, docker will fail to find them in the etcd docker container /etcd/passed file, and will fail to run the container. However, if we use an integer value, docker will be able use it and run the container even if that uid/gid doesn't map to an actual user inside the container. That's why we can only use the integer values here.

**Comment by alena1108** [comment]
LGTM

**Pull request merged**